### ISSUE TITLE
Issue 07: coarse dataset が optional trace-level usability labels を batch meta に載せる

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#368]
# Issue 07: coarse dataset が optional trace-level usability labels を batch meta に載せる

## 背景

Issue 06 で、OOF coarse error 由来の `coarse_usable_label` を含む NPZ を生成する。Step 3 の supervised usability head を学習するには、coarse training dataset がこの label を読み、anchor view に射影して batch meta に載せる必要がある。

現状 `GlobalAnchorCoarseDataset` は `fb_files` を file_info 経由で読み、`fb_idx_view` を meta に載せている。これと同様に、optional な trace-level label を扱う。

## ゴール

config で指定された label files を読み、training / labeled infer batch の `batch['meta']` に以下を追加する。

```text
coarse_usable_label_view: Tensor[B,H] or ndarray[H]
```

値は `1`, `0`, `-1`。

## 変更対象の候補

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/config.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/train.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/build_dataset.py`

## config 案

`paths` に optional list を追加する。

```yaml
paths:
  segy_files: ...
  fb_files: ...
  coarse_usability_label_files: ...        # optional, train files と同じ長さ
  infer_coarse_usability_label_files: ...  # optional, infer files と同じ長さ
```

または、`trace_labels` section にする案でもよい。

```yaml
trace_labels:
  coarse_usability:
    train_files: ...
    infer_files: ...
    label_key: coarse_usable_label
```

既存 `paths` との整合を考えると、最初は `paths.coarse_usability_label_files` を推奨する。

### 1. config load

`CoarsePaths` に optional fields を追加する。

```python
coarse_usability_label_files: tuple[str, ...] | None
infer_coarse_usability_label_files: tuple[str, ...] | None
```

validation:

- 指定されている場合、対応する `segy_files` / `infer_segy_files` と同じ長さ。
- train usability loss が有効なのに label files が無い場合は config error。
- usability loss disabled なら label files は無くてよい。

### 2. dataset init で labels を読む

`GlobalAnchorCoarseDataset.__init__()` に optional `coarse_usability_label_files` を渡す。

実装案:

- file_info と同じ順番で `np.load(label_file)` を読み込む。
- key `coarse_usable_label` を必須にする。
- dtype は int8 または integer を受け、内部で `np.int8` にする。
- shape は `n_traces` と一致すること。
- 値は `{-1,0,1}` のみ許可。
- `self._coarse_usable_labels_by_file_idx` のように保持する。

### 3. sample build 時に anchor view へ載せる

`_build_global_anchor_sample()` で、選ばれた full gather indices に対応する label を取り出す。

流れ:

```python
labels_full = label_array[indices_full]
labels_for_anchors = np.full((trace_len,), -1, dtype=np.int8)
labels_for_anchors[trace_valid] = labels_full[anchor_source_pos[trace_valid]]
```

augment / hflip / trace direction transform がある場合は、`fb_idx_view` と同じ trace projection を適用する必要がある。

推奨:

- trace label 専用 helper `project_trace_label_view(labels, H, meta)` を追加する。
- time 方向の `factor/start` は関係しない。
- `hflip` と `factor_h` のみ反映する。
- invalid / padding は `-1`。

最低限、現状の `trace_valid_view` と同じ trace axis projection になるようにする。

### 4. output meta に入れる

`meta` に以下を追加する。

```python
"coarse_usable_label_view": coarse_usable_label_view
```

`build_output_base()` / default collate により Tensor 化されることを確認する。必要なら `extra` にも入れる。

## 受け入れ条件

- label files 未指定かつ usability disabled では既存 dataset と同じ。
- label files 指定時、batch meta に `coarse_usable_label_view` が入る。
- shape は `(H,)`、DataLoader collate 後は `(B,H)`。
- 値は `-1/0/1` のみ。
- hflip など trace 方向 augment が入っても、label と trace_valid の位置が一致する。
- shape mismatch / invalid value / file count mismatch で明確なエラー。

## テスト

新規テスト候補:

- `tests/test_fbpick_coarse_usability_labels_dataset.py`

実データを使わない unit test が難しければ、projection helper の pure function test を優先する。

テスト内容:

1. label projection no augment。
2. hflip 時に label が反転する。
3. invalid value でエラー。
4. config file count mismatch でエラー。

## 非ゴール

- OOF label 生成は Issue 06。
- usability supervised loss は Issue 08。
- STALTA/AIC/cross-correlation の features は追加しない。
